pkg/auth/oauth/handlers/interfaces.go

```diff
@@ -32,6 +32,15 @@ type AuthenticationErrorHandler interface {
 	AuthenticationError(error, http.ResponseWriter, *http.Request) (handled bool, err error)
 }
 
+type ProviderInfo struct {
+	ID  string
+	URL string
+}
+
+type AuthenticationSelectionHandler interface {
+	SelectAuthentication([]ProviderInfo, http.ResponseWriter, *http.Request) (selected ProviderInfo, handled bool, err error)
```

Comment:
doc the return values... what would you think of returning *ProviderInfo so the consumer can check if it is nil rather than `len(provider.Name) == 0`?